adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralInterstitialAdLoader.m

```diff
@@ -0,0 +1,121 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterMintegralInterstitialAdLoader.h"
+#import "GADMAdapterMintegralUtils.h"
+#import "GADMediationAdapterMintegralConstants.h"
+#import "GADMAdapterMintegralExtras.h"
+
+#include <stdatomic.h>
+#import <MTGSDKNewInterstitial/MTGSDKNewInterstitial.h>
+#import <MTGSDKNewInterstitial/MTGNewInterstitialBidAdManager.h>
+
+@interface GADMAdapterMintegralInterstitialAdLoader ()<MTGNewInterstitialBidAdDelegate>
+
+@end
+
+@implementation GADMAdapterMintegralInterstitialAdLoader {
+    /// The completion handler to call when the ad loading succeeds or fails.
+    GADMediationInterstitialLoadCompletionHandler _adLoadCompletionHandler;
+    
+    /// Ad configuration for the ad to be loaded.
+    GADMediationInterstitialAdConfiguration *_adConfiguration;
+    
+    /// The Mintegral interstitial ad.
+    MTGNewInterstitialBidAdManager *_interstitialAd;
+    
+    /// The ad event delegate to forward ad rendering events to the Google Mobile Ads SDK.
+    id<GADMediationInterstitialAdEventDelegate> _adEventDelegate;
+}
+
+- (void)loadInterstitialForAdConfiguration:(nonnull GADMediationInterstitialAdConfiguration *)adConfiguration completionHandler:(nonnull GADMediationInterstitialLoadCompletionHandler)
+                                                 completionHandler{
+    _adConfiguration = adConfiguration;
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block GADMediationInterstitialLoadCompletionHandler originalCompletionHandler =
+    [completionHandler copy];
+    _adLoadCompletionHandler = ^id<GADMediationInterstitialAdEventDelegate>(
+                                                                        _Nullable id<GADMediationInterstitialAd> ad, NSError *_Nullable error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationInterstitialAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+    
+    NSString *adUnitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    if (!adUnitId.length || !placementId.length) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID or Placement ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+
+    _interstitialAd = [[MTGNewInterstitialBidAdManager alloc]initWithPlacementId:placementId unitId:adUnitId delegate:self];
+    [_interstitialAd loadAdWithBidToken:adConfiguration.bidResponse];
+}
+
```

Comment:
Can you add a `#pragma mark - MTGNewInterstitialBidAdDelegate`?

Reply:
OK, I've made changes and committed.